Confidence/src/test/java/com/spotify/confidence/EventSenderIntegrationTest.kt

```diff
@@ -118,29 +118,34 @@ class EventSenderIntegrationTest {
                 return false
             }
         }
+        val debugLogger = DebugLoggerMock()
         val engine = EventSenderEngineImpl(
             eventStorage,
             clientSecret,
             flushPolicies = mutableListOf(flushPolicy),
             dispatcher = testDispatcher,
             sdkMetadata = SdkMetadata("kotlin_test", ""),
-            uploader = uploader
+            uploader = uploader,
+            debugLogger = debugLogger
         )
         eventSender = Confidence(
             eventSenderEngine = engine,
             dispatcher = testDispatcher,
             diskStorage = mock(),
             clientSecret = "",
             flagApplierClient = mock(),
-            flagResolver = mock()
+            flagResolver = mock(),
+            debugLogger = debugLogger
         )
         val eventSender = this@EventSenderIntegrationTest.eventSender
         val eventCount = 4 * batchSize + 2
         requireNotNull(eventSender)
         repeat(eventCount) {
             eventSender.track("navigate")
         }
+        Assert.assertEquals(eventCount * 4, debugLogger.eventsLogged)
         advanceUntilIdle()
+        Assert.assertEquals(18, debugLogger.messagesLogged)
```

Comment:
what is 18?

Reply:
Same here, .messagesLogged isn't a unique log, but is triggered differently depending on the event. I was debating having it in the tests in general, since it's an arbitrary number, but decided to add and discuss if we want to keep it. I would like to keep it for the test purposes tbh. WDYT?

Reply:
one way we could test this more deterministically was to introduce the type in logEvent, like, `logEvent(EventType.FlushTriggered ...)` or something like this, then we could test this better. I like to keep the test but somehow comment something or make it clear what this 18 means.

Reply:
I'll add a comment!